packages/flutter_tools/lib/src/commands/apk.dart

```diff
@@ -0,0 +1,201 @@
+  static const String _kBuildToolsVersion = '22.0.1';
+
+  final String androidSdk;
+
+  File _androidJar;
+  File _aapt;
+  File _zipalign;
+  String _jarsigner;
+
+  ApkBuilder(this.androidSdk) {
+    _androidJar = new File('$androidSdk/platforms/android-$_kAndroidPlatformVersion/android.jar');
+
+    String buildTools = '$androidSdk/build-tools/$_kBuildToolsVersion';
+    _aapt = new File('$buildTools/aapt');
+    _zipalign = new File('$buildTools/zipalign');
+    _jarsigner = 'jarsigner';
+  }
+
+  void package(File outputApk, File androidManifest, Directory assets, Directory artifacts) {
+    _run(_aapt.path, [
+      'package',
+      '-M', androidManifest.path,
+      '-A', assets.path,
+      '-I', _androidJar.path,
+      '-F', outputApk.path,
+      artifacts.path
+    ]);
+  }
+
+  void sign(File keystore, String keystorePassword, String keyName, File outputApk) {
+    _run(_jarsigner, [
+      '-keystore', keystore.path,
+      '-storepass', keystorePassword,
+      outputApk.path,
+      keyName,
+    ]);
+  }
+
+  void align(File unalignedApk, File outputApk) {
+    _run(_zipalign.path, ['-f', '4', unalignedApk.path, outputApk.path]);
+  }
+
+  void _run(String command, List<String> args, { String workingDirectory }) {
+    ProcessResult result = Process.runSync(
+        command, args, workingDirectory: workingDirectory);
+    if (result.exitCode == 0)
+      return;
+    stdout.write(result.stdout);
+    stderr.write(result.stderr);
+  }
+}
+
+class ApkCommand extends FlutterCommand {
+  final String name = 'apk';
+  final String description = 'Build an Android APK package.';
+
+  ApkCommand() {
+    argParser.addOption('manifest',
+        abbr: 'm',
+        defaultsTo: _kDefaultAndroidManifestPath,
+        help: 'Android manifest XML file.');
+    argParser.addOption('output-file',
+        abbr: 'o',
+        defaultsTo: _kDefaultOutputPath,
+        help: 'Output APK file.');
+    argParser.addOption('target',
+        abbr: 't',
+        defaultsTo: '',
+        help: 'Target app path or filename used to build the FLX..');
+    argParser.addOption('flx',
+        abbr: 'f',
+        defaultsTo: '',
+        help: 'Path to the FLX file. If this is not provided, an FLX will be built.');
+  }
+
+  void _buildApk(String flxPath) {
+    File androidManifest = new File(argResults['manifest']);
+    File icuData = new File('${runner.enginePath}/third_party/icu/android/icudtl.dat');
+    File classesDex = new File('${runner.enginePath}/out/android_Debug/gen/sky/shell/shell/classes.dex');
+    File libSkyShell = new File('${runner.enginePath}/out/android_Debug/gen/sky/shell/shell/shell/libs/armeabi-v7a/libsky_shell.so');
+    File keystore = new File('${runner.enginePath}/build/android/ant/chromium-debug.keystore');
+
+    for (File f in [androidManifest, icuData, classesDex, libSkyShell, keystore]) {
+      if (!f.existsSync()) {
+        _logging.severe('Can not locate file: ${f.path}');
+        return;
```

Comment:
Should we return some sort of failure indication here so the tool can exit with a non-zero exit code?